src/beamlime/applications/daemons.py

```diff
@@ -0,0 +1,173 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2024 Scipp contributors (https://github.com/scipp)
+import asyncio
+from contextlib import contextmanager
+from typing import Any, Awaitable, Callable, Coroutine, Generator, List, Optional
+
+from ._parameters import ChunkSize, DataFeedingSpeed
+from ._random_data_providers import RandomEvents
+from ._workflow import Events
+from .base import BeamlimeMessage, DaemonInterface
+from .handlers import RawDataSent
+
+
+class MessageRouter(DaemonInterface):
+    """A message router that routes messages to handlers."""
+
+    message_pipe: List[BeamlimeMessage]
```

Comment:
Shouldn't this be a queue? (https://docs.python.org/3/library/queue.html)
Also, you don't need a class annotation for it.